FlyingSocks/Sources/Socket.swift

```diff
@@ -53,11 +69,11 @@ public struct Socket: Sendable, Hashable {
     }
 
     public init(domain: Int32) throws {
-        try self.init(domain: domain, type: Socket.stream)
+        try self.init(domain: domain, type: .stream)
     }
 
-    public init(domain: Int32, type: Int32) throws {
-        let descriptor = FileDescriptor(rawValue: Socket.socket(domain, type, 0))
+    public init(domain: Int32, type: SocketType) throws {
```

Comment:
Because this is public API we should probably leave the existing method but deprecate it

```swift
@available(*, deprecated, message: "type is now SocketType")
public init(domain: Int32, type: Int32) throws {
```